src/uiprotect/api.py

```diff
@@ -1482,7 +1487,150 @@ async def _stream_response(
+
+                await self._stream_response(r, chunk_size, callback, progress_callback)
+        else:
+            await self._stream_response(
+                r,
+                chunk_size,
+                iterator_callback,
+                progress_callback,
+            )
+        r.close()
+        return None
+
+    async def _validate_channel_id(self, camera_id: str, channel_index: int) -> None:
+        """Validate if camera_id and channel_index are valid."""
+        if self._bootstrap is None:
+            await self.update()
+        try:
+            camera = self._bootstrap.cameras[camera_id]  # type: ignore[union-attr]
+            camera.channels[channel_index]
+        except (IndexError, AttributeError, KeyError) as e:
+            raise BadRequest(f"Invalid input: {e}") from e
+
+    async def prepare_camera_video(
+        self,
+        camera_id: str,
+        start: datetime,
+        end: datetime,
+        channel_index: int = 0,
+        validate_channel_id: bool = True,
+        fps: int | None = None,
+        filename: str | None = None,
+    ) -> list[Any] | dict[str, Any] | None:
+        """
+        Prepares MP4 video from a given camera at a specific time.
+
+        This is the newer implementation of retrieving video from Unifi Protect.
+        When using this method, it should be followed up with download_camera_video().
+
+        Start/End of video export are approximate. It may be +/- a few seconds.
+
+        Providing the `fps` parameter creates a "timelapse" export wtih the given FPS
+        value. Protect app gives the options for 60x (fps=4), 120x (fps=8), 300x
+        (fps=20), and 600x (fps=40).
+
+        You will receive a filename and an expiry time in seconds.
+        """
+        if self.bootstrap.nvr.version < NEW_DOWNLOAD_VERSION:
+            raise ValueError(
+                "This method is only support from Unifi Protect version >= 4.0.0."
+            )
+
+        if validate_channel_id:
+            await self._validate_channel_id(camera_id, channel_index)
+
+        params = {
+            "camera": camera_id,
+            "start": to_js_time(start),
+            "end": to_js_time(end),
+        }
+
+        if channel_index == 3:
+            params.update({"lens": 2})
+        else:
+            params.update({"channel": channel_index})
+
+        if fps is not None and fps > 0:
+            params["fps"] = fps
+            params["type"] = VideoExportType.TIMELAPSE
+        else:
+            params["type"] = VideoExportType.ROTATING
+
+        if not filename:
+            start_str = start.strftime("%m-%d-%Y, %H.%M.%S %Z")
+            end_str = end.strftime("%m-%d-%Y, %H.%M.%S %Z")
+            filename = f"{camera_id} {start_str} - {end_str}.mp4"
+
+        params["filename"] = filename
+
+        path = "video/prepare"
+
+        return await self.api_request(
+            path,
+            params=params,
+            raise_exception=True,
+        )
```

Comment:
```suggestion
        return await self.api_request(
            "video/prepare",
            params=params,
            raise_exception=True,
        )
```

Unnecessary assignment to temp variable